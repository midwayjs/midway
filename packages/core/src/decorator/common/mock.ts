import { Scope, Provide, saveModule, MOCK_KEY } from '../';
import { ScopeEnum } from '../../interface';

export function Mock(): ClassDecorator {
  return (target: any) => {
    saveModule(MOCK_KEY, target);
    Scope(ScopeEnum.Singleton)(target);
    Provide()(target);
  };
}
