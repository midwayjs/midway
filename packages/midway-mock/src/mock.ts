import * as mock from 'egg-mock';
import { resolveModule } from 'midway-bin';
import { join } from 'path';
import { MidwayApplicationOptions, MidwayMockApplication } from './interface';

export interface MidwayMock extends mock.EggMock {
  container: typeof mockContainer;
  default: mock.EggMock;
  app: (option?: MidwayApplicationOptions) => MidwayMockApplication;
  cluster: (option?: MidwayApplicationOptions) => MidwayMockApplication;
}

/**
 * 只初始化app级别的container
 * agent相关的逻辑使用mm2.app
 * @param options 参数
 */
function mockContainer(options: MidwayApplicationOptions): MockContainer {
  return new MockContainer(options);
}

const getDefaultFramework: () => string = () => resolveModule('midway') || resolveModule('midway-mirror');

export const mm = Object.assign({}, mock, {
  container: mockContainer,
}) as MidwayMock;

mm.app = (options = {}): MidwayMockApplication => {
  if (process.env.MIDWAY_BASE_DIR && !options.baseDir) { options.baseDir = process.env.MIDWAY_BASE_DIR; }
  if (process.env.MIDWAY_FRAMEWORK_PATH && !options.framework) { options.framework = process.env.MIDWAY_FRAMEWORK_PATH; }
  // @ts-ignore
  return mock.app(Object.assign({
    framework: options.framework || getDefaultFramework(),
    typescript: !!require.extensions['.ts'],
    plugins: {
      'midway-mock': {
        path: join(__dirname, '../')
      }
    }
  }, options));
};

mm.cluster = (options = {}) => {
  if (process.env.MIDWAY_BASE_DIR && !options.baseDir) { options.baseDir = process.env.MIDWAY_BASE_DIR; }
  if (process.env.MIDWAY_FRAMEWORK_PATH && !options.framework) { options.framework = process.env.MIDWAY_FRAMEWORK_PATH; }
  // @ts-ignore
  return mock.cluster(Object.assign({
    framework: options.framework || getDefaultFramework(),
    typescript: !!require.extensions['.ts'],
    plugins: {
      'midway-mock': {
        path: join(__dirname, '../')
      }
    }
  }, options));
};

export class MockContainer {

  app: MidwayMockApplication;

  constructor(options: MidwayApplicationOptions = {}) {
    this.app = mm.app(options);
  }

  async ready() {
    await this.app.ready();
  }

  async getAsync(id: any) {
    return this.app.applicationContext.getAsync(id);
  }

  get(id: any) {
    return this.app.applicationContext.get(id);
  }
}
