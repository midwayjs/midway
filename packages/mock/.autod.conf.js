'use strict';

module.exports = {
  write: true,
  prefix: '^',
  devprefix: '^',
  exclude: [
    'test/fixtures',
    'examples',
    'docs',
    'run',
  ],
  dep: [
  ],
  devdep: [
  ],
  keep: [
  ]
};
