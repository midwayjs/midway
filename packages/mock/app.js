const { createApp, close, processArgsParser } = require('./dist');
const { join } = require('path');

(async () => {
  process.env.MIDWAY_TS_MODE = 'false';
  const args = processArgsParser(process.argv);
  if (args.port) {
    process.env.MIDWAY_HTTP_PORT = args.port;
  }

  process.once('SIGINT', onSignal);
  // kill(3) Ctrl-\
  process.once('SIGQUIT', onSignal);
  // kill(15) default
  process.once('SIGTERM', onSignal);

  let app = undefined;
  app = await createApp({
    appDir: process.cwd(),
    baseDir: join(process.cwd(), 'dist'),
    ...args,
  });

  process.send({
    title: 'server-ready',
    port: process.env.MIDWAY_HTTP_PORT,
    ssl: args.ssl,
  });

  function onSignal() {
    close(app).then(() => {
      process.exit(0);
    });
  }
})();
