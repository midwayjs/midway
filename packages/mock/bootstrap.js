'use strict';

// exports = require('./dist/bootstrap')

function __export(m) {
  // eslint-disable-next-line no-prototype-builtins
  for (const p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, '__esModule', { value: true });
__export(require('./dist/bootstrap'));
