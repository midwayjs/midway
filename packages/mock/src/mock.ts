import {
  getCurrentMainApp,
  IMidwayApplication,
  IMidwayContext,
  MidwayMockService,
} from '@midwayjs/core';

function getMockService(app?): MidwayMockService {
  if (!app) {
    app = getCurrentMainApp();
  }
  if (!app) {
    return null;
  }
  if (!app.getApplicationContext) {
    throw new Error('[mock]: app.getApplicationContext is undefined.');
  }

  const applicationContext = app.getApplicationContext();
  const mockService = applicationContext.get(MidwayMockService);
  if (!mockService) {
    throw new Error('[mock]: MidwayMockService is undefined.');
  }
  return mockService;
}

export function mockSession(
  app: IMidwayApplication<any>,
  key: string,
  value: any
) {
  const mockService = getMockService(app);
  mockService.mockContext(app, (ctx: any) => {
    if (!ctx.session) {
      ctx.session = {};
    }
    ctx.session[key] = value;
  });
}

export function mockHeader(
  app: IMidwayApplication<any>,
  headerKey: string,
  headerValue: string
) {
  const mockService = getMockService(app);
  mockService.mockContext(app, (ctx: any) => {
    ctx.headers[headerKey] = headerValue;
  });
}

export function mockClassProperty(
  clzz: new (...args) => any,
  propertyName: string,
  value: any
) {
  const mockService = getMockService();
  if (!mockService) {
    return MidwayMockService.mockClassProperty(clzz, propertyName, value);
  } else {
    return mockService.mockClassProperty(clzz, propertyName, value);
  }
}

export function mockProperty(obj: any, key: string, value) {
  const mockService = getMockService();
  if (!mockService) {
    return MidwayMockService.mockProperty(obj, key, value);
  } else {
    return mockService.mockProperty(obj, key, value);
  }
}

export function restoreAllMocks() {
  const mockService = getMockService();
  if (mockService) {
    mockService.restore();
  } else {
    MidwayMockService.prepareMocks = [];
  }
}

export function mockContext(
  app: IMidwayApplication<any>,
  key: string | ((ctx: IMidwayContext) => void),
  value?: PropertyDescriptor | any
) {
  const mockService = getMockService(app);
  mockService.mockContext(app, key, value);
}
