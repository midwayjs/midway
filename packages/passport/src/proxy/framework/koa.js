'use strict'

/**
 * Module dependencies.
 */
const passport = require('passport')

/**
 * Passport's default/connect middleware.
 */
const _initialize   = require('passport/lib/middleware/initialize')
const _authenticate = require('passport/lib/middleware/authenticate')
const createReqMock = require('./request').create

/**
 * Passport's initialization middleware for Koa.
 *
 * @return {GeneratorFunction}
 * @api private
 */
function initialize(passport) {
  const middleware = promisify(_initialize(passport))
  return function passportInitialize(ctx, next) {
    // koa <-> connect compatibility:
    const userProperty = passport._userProperty || 'user'
    // check ctx.req has the userProperty
    if (!ctx.req.hasOwnProperty(userProperty)) {
      Object.defineProperty(ctx.req, userProperty, {
        enumerable: true,
        get: function() {
          return ctx.state[userProperty]
        },
        set: function(val) {
          ctx.state[userProperty] = val
        }
      })
    }

    // create mock object for express' req object
    const req = createReqMock(ctx, userProperty)


    // add Promise-based login method
    const login  = req.login
    ctx.login = ctx.logIn = function(user, options) {
      return new Promise((resolve, reject) => {
        login.call(req, user, options, err => {
          if (err) reject(err)
          else resolve()
        })
      })
    }

    // add aliases for passport's request extensions to Koa's context
    ctx.logout = ctx.logOut = req.logout.bind(req)
    ctx.isAuthenticated     = req.isAuthenticated.bind(req)
    ctx.isUnauthenticated   = req.isUnauthenticated.bind(req)

    return middleware(req, ctx).then(function() {
      return next()
    })
  }
}

/**
 * Passport's authenticate middleware for Koa.
 *
 * @param {String|Array} name
 * @param {Object} options
 * @param {GeneratorFunction} callback
 * @return {GeneratorFunction}
 * @api private
 */
function authenticate(passport, name, options, callback) {
  // normalize arguments
  if (typeof options === 'function') {
    callback = options
    options  = {}
  }
  options = options || {}

  if (callback) {
    // When the callback is set, neither `next`, `res.redirect` or `res.end`
    // are called. That is, a workaround to catch the `callback` is required.
    // The `passportAuthenticate()` method below will therefore set
    // `callback.resolve` and `callback.reject`. Then, once the authentication
    // finishes, the modified callback calls the original one and afterwards
    // triggers either `callback.resolve` or `callback.reject` to inform
    // `passportAuthenticate()` that we are ready.
    const _callback = callback
    callback = function(err, user, info, status) {
      try {
        Promise.resolve(_callback(err, user, info, status))
               .then(() => callback.resolve(false))
               .catch(err => callback.reject(err))
      } catch (err) {
        callback.reject(err)
      }
    }
  }

  const middleware = promisify(_authenticate(passport, name, options, callback))

  return function passportAuthenticate(ctx, next) {
    // this functions wraps the connect middleware
    // to catch `next`, `res.redirect` and `res.end` calls
    const p = new Promise((resolve, reject) => {
      // mock the `req` object
      const req = createReqMock(ctx, options.assignProperty || passport._userProperty || 'user')

      function setBodyAndResolve(content) {
        if (content) ctx.body = content
        resolve(false)
      }

      // mock the `res` object
      const res = {
        redirect: function(url) {
          ctx.redirect(url)
          resolve(false)
        },
        set: ctx.set.bind(ctx),
        setHeader: ctx.set.bind(ctx),
        end: setBodyAndResolve,
        send: setBodyAndResolve,
        set statusCode(status) {
          ctx.status = status
        },
        get statusCode() {
          return ctx.status
        }
      }

      req.res = res

      // update the custom callback above
      if (callback) {
        callback.resolve = resolve
        callback.reject  = reject
      }

      // call the connect middleware
      middleware(req, res).then(resolve, reject)
    })

    return p.then(cont => {
      // cont equals `false` when `res.redirect` or `res.end` got called
      // in this case, call next to continue through Koa's middleware stack
      if (cont !== false) {
        return next()
      }
    })
  }
}

/**
 * Passport's authorize middleware for Koa.
 *
 * @param {String|Array} name
 * @param {Object} options
 * @param {GeneratorFunction} callback
 * @return {GeneratorFunction}
 * @api private
 */
function authorize(passport, name, options, callback) {
  options = options || {}
  options.assignProperty = 'account'

  return authenticate(passport, name, options, callback)
}

/**
 * Framework support for Koa.
 *
 * This module provides support for using Passport with Koa. It exposes
 * middleware that conform to the `fn*(next)` signature and extends
 * Node's built-in HTTP request object with useful authentication-related
 * functions.
 *
 * @return {Object}
 * @api protected
 */
module.exports = function() {
  return {
    initialize:   initialize,
    authenticate: authenticate,
    authorize:    authorize
  }
}

function promisify(expressMiddleware) {
  return function(req, res) {
    return new Promise(function(resolve, reject) {
      expressMiddleware(req, res, function(err, result) {
        if (err) reject(err)
        else resolve(result)
      })
    })
  }
}
