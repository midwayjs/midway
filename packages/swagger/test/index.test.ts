import { close, createApp, createHttpRequest } from '@midwayjs/mock';
import * as koa from '@midwayjs/koa';
import { join } from 'path';

describe('/test/index.test.ts', () => {

  describe('test swagger', () => {

    let app;
    beforeAll(async () => {
      try {
        app = await createApp(join(__dirname, 'fixtures/cats'), {}, koa);
      } catch (e) {
        console.log(e);
        console.log('beforeAll: ' +  e.stack);
      }
    });

    afterAll(async () => {
      await close(app);
    });

    it('should get swagger json', async () => {
      let ret = await createHttpRequest(app).get('/swagger-ui/index.html');
      expect(ret.type).toEqual('text/html');
      expect(ret.text).toContain('html');

      ret = await createHttpRequest(app).get('/swagger-ui/swagger-ui.js');
      expect(ret.type).toEqual('application/javascript');
      expect(ret.text).toContain('!function');

      ret = await createHttpRequest(app).get('/swagger-ui/swagger-ui.css');
      expect(ret.type).toEqual('text/css');
      expect(ret.text).toContain('.swagger-ui{color:#3b4151');

      const result = await createHttpRequest(app).get('/swagger-ui/index.json');
      expect(result.type).toEqual('application/json');
      const body = result.body;
      console.log('--->', JSON.stringify(body));

      expect(body.tags.length).toBeGreaterThanOrEqual(2);
      expect(body.tags).toStrictEqual([{"name":"1-你好这里","description":""},{"name":"2-国家测试","description":""},{"description":"","name":"sss"}]);
      expect(body.components.securitySchemes).toStrictEqual({"bbb":{"type":"http","scheme":"basic"},"ttt":{"type":"http","scheme":"bearer","bearerFormat":"JWT"}})
      expect(body.components.schemas.CatT).toMatchSnapshot();
      expect(body.components.schemas.Catd).toMatchSnapshot();
      expect(body.components.schemas.Cata).toMatchSnapshot();
      expect(body.components.schemas.Cat).toMatchSnapshot();
      expect(body.components.schemas.CreateCatDto).toMatchSnapshot();
      expect(body.paths['/cats/{id}']).toMatchSnapshot();
    });
  });

  it('should fix issue1976', async () => {
    const app = await createApp(join(__dirname, 'fixtures/issue1976'), {});
    const result = await createHttpRequest(app).get('/swagger-ui/index.json');
    expect(result.type).toEqual('application/json');
    const body = result.body;
    expect(body).toMatchSnapshot();
    console.log(JSON.stringify(body));
    await close(app);
  });

  it('should fix issue2603', async () => {
    const app = await createApp(join(__dirname, 'fixtures/issue2603'), {});
    const result = await createHttpRequest(app).get('/swagger-ui/index.json');
    expect(result.type).toEqual('application/json');
    const body = result.body;
    expect(body).toMatchSnapshot();
    console.log(JSON.stringify(body));
    await close(app);
  });
});
