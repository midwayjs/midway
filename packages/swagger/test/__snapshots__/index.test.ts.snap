// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/index.test.ts should fix issue1976 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "CodeLoginDTO": Object {
        "properties": Object {
          "code": Object {
            "description": "验证码",
            "example": "0000",
            "type": "string",
          },
          "phone": Object {
            "description": "11位手机号码",
            "example": "17755556666",
            "type": "string",
          },
          "site": Object {
            "description": "产品线",
            "enum": Array [
              "sport",
              "question",
            ],
            "type": "string",
          },
          "uuid": Object {
            "description": "随机数",
            "example": "xxxx",
            "type": "string",
          },
        },
        "type": "object",
      },
      "LoginDTO": Object {
        "properties": Object {
          "password": Object {
            "description": "密码",
            "example": "xxx",
            "type": "string",
          },
          "phone": Object {
            "description": "11位手机号码",
            "example": "17755556666",
            "type": "string",
          },
          "site": Object {
            "description": "产品线",
            "enum": Array [
              "sport",
              "question",
            ],
            "type": "string",
          },
        },
        "type": "object",
      },
    },
  },
  "info": Object {
    "contact": Object {},
    "description": "This is a swagger-ui for midwayjs project",
    "title": "My Project",
    "version": "1.0.0",
  },
  "openapi": "3.0.1",
  "paths": Object {
    "/cats/login": Object {
      "post": Object {
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/LoginDTO",
              },
            },
          },
          "description": "",
          "required": true,
        },
        "responses": Object {
          "200": Object {
            "description": "OK",
          },
        },
        "tags": Array [
          "cats",
        ],
      },
    },
    "/cats/smsLogin": Object {
      "post": Object {
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/CodeLoginDTO",
              },
            },
          },
          "description": "",
          "required": true,
        },
        "responses": Object {
          "200": Object {
            "description": "OK",
          },
        },
        "tags": Array [
          "cats",
        ],
      },
    },
  },
  "servers": Array [],
  "tags": Array [
    Object {
      "description": "cats",
      "name": "cats",
    },
  ],
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 1`] = `
Object {
  "properties": Object {
    "agedt": Object {
      "description": "The age of ttthe Cat",
      "example": 1,
      "type": "number",
    },
    "breedtd": Object {
      "description": "The breed tttof the Cat",
      "example": "Maine Coon",
      "type": "string",
    },
    "namedt": Object {
      "description": "The name of the Cat",
      "example": "Kitty",
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 2`] = `
Object {
  "properties": Object {
    "aged": Object {
      "description": "The age of the Cat",
      "example": 1,
      "type": "number",
    },
    "breedd": Object {
      "description": "The breed of the Cat",
      "example": "Maine Coon",
      "type": "string",
    },
    "named": Object {
      "description": "The name of the Cat",
      "example": "Kitty",
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 3`] = `
Object {
  "properties": Object {
    "agea": Object {
      "description": "The age of thaae Cat",
      "example": 1,
      "type": "number",
    },
    "breeda": Object {
      "description": "The breed of taahe Cat",
      "example": "Maine Coon",
      "type": "string",
    },
    "catdsa": Object {
      "description": "The breed of the Cat",
      "items": Object {
        "$ref": "#/components/schemas/Catd",
      },
      "type": "array",
    },
    "namea": Object {
      "description": "The name aaof the Cat",
      "example": "Kitty",
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 4`] = `
Object {
  "properties": Object {
    "age": Object {
      "description": "The age of the Cat",
      "example": 1,
      "type": "number",
    },
    "agedata": Object {
      "description": "The age of the CatDSate",
      "example": "2022-12-12 11:11:11",
      "format": "date",
      "type": "string",
    },
    "breed": Object {
      "description": "The breed of the Cat",
      "example": "Maine Coon",
      "type": "string",
    },
    "catds": Object {
      "description": "The breed of the Cat",
      "items": Object {
        "$ref": "#/components/schemas/Catd",
      },
      "type": "array",
    },
    "name": Object {
      "description": "The name of the Cat",
      "example": "Kitty",
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 5`] = `
Object {
  "properties": Object {
    "age": Object {
      "description": "The name of the Catage",
      "example": "1",
      "maxItems": 1,
      "maxLength": 1,
      "maxProperties": 2,
      "maximum": 10,
      "type": "number",
    },
    "agedata": Object {
      "description": "The age of the CatDSate",
      "example": "2022-12-12 11:11:11",
      "format": "date",
      "type": "string",
    },
    "breed": Object {
      "description": "The name of the Catbreed",
      "example": "bbbb",
      "type": "string",
    },
    "breeds": Object {
      "description": "The name of the Catage",
      "example": Array [
        "1",
      ],
      "items": Object {
        "type": "string",
      },
      "nullable": true,
      "type": "array",
      "uniqueItems": true,
    },
    "hello": Object {
      "enum": Array [
        "One",
        "Two",
        "Three",
      ],
      "type": "string",
    },
    "name": Object {
      "description": "The name of the Catname",
      "example": "Kitty",
      "format": "binary",
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`/test/index.test.ts test swagger should get swagger json 6`] = `
Object {
  "get": Object {
    "parameters": Array [
      Object {
        "in": "header",
        "name": "x-test-one",
        "schema": Object {
          "type": "string",
        },
      },
      Object {
        "allowEmptyValue": false,
        "explode": true,
        "in": "query",
        "name": "aa",
        "required": true,
        "schema": Object {
          "enum": Array [
            "One",
            "Two",
            "Three",
          ],
          "type": "string",
        },
        "style": "form",
      },
      Object {
        "allowEmptyValue": false,
        "explode": true,
        "in": "query",
        "name": "test",
        "required": true,
        "schema": Object {
          "enum": Array [
            "One",
            "Two",
            "Three",
          ],
          "type": "string",
        },
        "style": "form",
      },
      Object {
        "description": "hello world id number",
        "example": 12,
        "explode": false,
        "in": "path",
        "name": "id",
        "required": true,
        "schema": Object {
          "type": "string",
        },
        "style": "simple",
      },
    ],
    "responses": Object {
      "200": Object {
        "content": Object {
          "application/json": Object {
            "schema": Object {
              "$ref": "#/components/schemas/Cat",
            },
          },
        },
        "description": "The found record",
      },
    },
    "security": Array [
      Object {
        "ttt": Array [],
      },
      Object {
        "bbb": Array [],
      },
    ],
    "tags": Array [
      "2-国家测试",
      "sss",
    ],
    "x-hello": Object {
      "hello": "world",
    },
  },
  "post": Object {
    "parameters": Array [
      Object {
        "in": "header",
        "name": "x-test-one",
        "schema": Object {
          "type": "string",
        },
      },
      Object {
        "description": "hello world id number",
        "example": 12,
        "explode": false,
        "in": "path",
        "name": "id",
        "required": true,
        "schema": Object {
          "format": "int32",
          "type": "number",
        },
        "style": "simple",
      },
    ],
    "requestBody": Object {
      "content": Object {
        "application/octet-stream": Object {
          "schema": Object {
            "format": "binary",
            "type": "string",
          },
        },
      },
      "description": "hello world",
      "required": true,
    },
    "responses": Object {
      "200": Object {
        "content": Object {
          "application/json": Object {
            "schema": Object {
              "$ref": "#/components/schemas/Cat",
            },
          },
        },
        "description": "成功了",
      },
      "403": Object {
        "description": "Forbidden.",
      },
      "404": Object {
        "description": "NotFound.",
      },
    },
    "security": Array [
      Object {
        "ttt": Array [],
      },
      Object {
        "bbb": Array [],
      },
    ],
    "summary": "Create cat",
    "tags": Array [
      "2-国家测试",
      "sss",
    ],
  },
}
`;
